src/Libraries/CoreNodes/List.cs

```diff
@@ -260,6 +260,38 @@ public static IList AddItemToEnd(object item, IList list)
                 item //Add item to the end of cloned list.
             };
         }
+        
+        /// <summary>
+        ///     Inserts an item into a list at a specified index.
+        /// </summary>
+        /// <param name="item">Item to be inserted.</param>
+        /// <param name="list">List to insert in to.</param>
+        /// <param name="index">Index to insert item at.</param>
+        /// <search>insert,add,item</search>
+        public static IList InsertItemAtIndex(object item, IList list, int index)
+        {
+            while (index < 0)
+                index += list.Count;
```

Comment:
I don't believe this is the correct behaviour.

{0, 2}[3] fails with an out of bounds error
  so
{0, 2}[-3] should also fail.